### ISSUE TITLE
Cannot load new libraries into Dynamo

### ISSUE THREAD

**Issue by Olesen666** [DynamoDS/Dynamo#9115]
## Dynamo version

Dynamo Core: 2.1.0.6255
Dynamo Revit 2.1.0.6256

## Operating system

Windows 10.

## What did you do? 

Clicked "File" -> "Import Library" and navigated to the package .dll

## What did you expect to see?

Expected to be able to import libraries e.g. packages into dynamo if these are not loaded automatically.

## What did you see instead?

An error message saying "Failed to load library: C:\....Appdata\Roaming\Dynamo\Dynamo Revit\2.1\packages\DynamoMEP\bin\DynamoMEP.dll

(Have tried with other packages too with same results).

![image](https://user-images.githubusercontent.com/43544154/45957536-88fe1800-c015-11e8-99b6-007ce1de8e1f.png)

Additional info:
Exception of type 'ProtoCore.Exceptions.CompilerInternalException' was thrown.

   at ProtoCore.AssociativeEngine.Utils.IsGlobalScopeDirty(Executable exe)
   at ProtoScript.Runners.LiveRunner.ApplyUpdate()
   at ProtoScript.Runners.LiveRunner.CompileAndExecuteForDeltaExecution(String code)
   at ProtoScript.Runners.LiveRunner.UpdateCmdLineInterpreter(String code)
   at Dynamo.Engine.EngineController.OnLibraryLoaded()
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at Dynamo.Engine.LibraryServices.OnLibraryLoaded(LibraryLoadedEventArgs e)
   at Dynamo.Engine.LibraryServices.ImportLibrary(String library, Boolean isExplicitlyImportedLib)
   at Dynamo.ViewModels.DynamoViewModel.ImportLibrary(Object parameter)
   at MS.Internal.Commands.CommandHelpers.CriticalExecuteCommandSource(ICommandSource commandSource, Boolean userInitiated)
   at System.Windows.Controls.MenuItem.InvokeClickAfterRender(Object arg)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by johnpierson** [comment]
@Olesen666, I noticed that you have the package loaded in the package directory already. Is it not loading as expected on Dynamo opening?

**Comment by johnpierson** [comment]
Odd, and have you published the package locally while including the DLL "As Node Library"

**Comment by Olesen666** [comment]
Hi John, you hit the nail on the head, the package was published (locallay) prior to the existence of the ZT nodes. I've just tried publishing a new version with the ".dll" as node library (btw that function was well hidden) but not it is loaded pr. default (awesome) Thanks a lot. 

The original issue was caused by the attempt of coding a "unzip" node which for some reason "broke" the package... So will attempt another approach. 

Again thanks for the replies, I'll close the ticket.
